### PR TITLE
Create 2020-07-10-article-on-reactive-programming.md

### PR DESCRIPTION
**Pull request by richbryant**
Added blog article

**What are the main goals of this change?**
- To start using the Blog section of ReactiveUI.net for actual articles so that we can bring in more users and explain things in simpler terms.  Basically, I wrote an origin story, feel free to say inappropriate if you like.

- [x ] New content

**Is this change related to any reported issue? (Optional)**

Nope.

**Notes (Optional)**

I plan to move on from this to real-world examples of Rx.net, ReactiveUI, and how FP actually works in C#.  Partial application and currying are badly in need of some non-mystical explanations.

### CODE REVIEW

**Comment by glennawatson** [comment]
Look at the other articles btw in that folder for the header format you should use

**Comment by richbryant** [comment]
Roger that, boss.
